	* glib-convert.adb, glib-module.adb, gdk-pixbuf.adb, gtk-accel_map.adb,
	gtk-file_chooser.adb, gtk-file_selection.adb, gtk-icon_factory.adb,
	gtk-icon_theme.adb, gtk-rc.adb, gtk-ui_manager.adb,
	gtk-window.adb: Fix import names for Windows.

	* generated/gtk-image.adb: Ditto.

Tue Jul  9 11:48:58 2002  Arnaud Charlet  <charlet@gnat.com>

	* glib-values.ads: Fix definition of GValue to conform to ms bitfields
	packing under Windows.

Index: gtk-file_chooser.adb
===================================================================
--- gtk-file_chooser.adb	(revision 174374)
+++ gtk-file_chooser.adb	(working copy)
@@ -60,7 +60,7 @@
          Folder  : String;
          Error   : access Glib.Error.GError)
          return Gboolean;
-      pragma Import (C, Internal, "gtk_file_chooser_add_shortcut_folder");
+      pragma Import (C, Internal, "gtk_file_chooser_add_shortcut_folder_utf8");
       Err : aliased Glib.Error.GError;
    begin
       if Internal (Chooser, Folder & ASCII.NUL, Err'Unchecked_Access) = 0 then
@@ -114,7 +114,7 @@
    function Get_Filename (Chooser : Gtk_File_Chooser) return String is
       function Internal
         (Chooser : Gtk_File_Chooser) return Interfaces.C.Strings.chars_ptr;
-      pragma Import (C, Internal, "gtk_file_chooser_get_filename");
+      pragma Import (C, Internal, "gtk_file_chooser_get_filename_utf8");
 
       Result : chars_ptr := Internal (Chooser);
    begin
@@ -138,7 +138,7 @@
      (Chooser : Gtk_File_Chooser) return Gtk.Enums.String_SList.GSlist
    is
       function Internal (Chooser : Gtk_File_Chooser) return System.Address;
-      pragma Import (C, Internal, "gtk_file_chooser_get_filenames");
+      pragma Import (C, Internal, "gtk_file_chooser_get_filenames_utf8");
       L : Gtk.Enums.String_SList.GSlist;
    begin
       Gtk.Enums.String_SList.Set_Object (L, Internal (Chooser));
@@ -151,7 +151,7 @@
 
    function Get_Current_Folder (Chooser : Gtk_File_Chooser) return String is
       function Internal (Chooser : Gtk_File_Chooser) return chars_ptr;
-      pragma Import (C, Internal, "gtk_file_chooser_get_current_folder");
+      pragma Import (C, Internal, "gtk_file_chooser_get_current_folder_utf8");
 
       Result : chars_ptr := Internal (Chooser);
    begin
@@ -238,7 +238,8 @@
    function Get_Preview_Filename (Chooser : Gtk_File_Chooser) return String
    is
       function Internal (Chooser : Gtk_File_Chooser) return chars_ptr;
-      pragma Import (C, Internal, "gtk_file_chooser_get_preview_filename");
+      pragma Import
+        (C, Internal, "gtk_file_chooser_get_preview_filename_utf8");
 
       Result : chars_ptr := Internal (Chooser);
    begin
@@ -401,7 +402,8 @@
      (Chooser : Gtk_File_Chooser) return Gtk.Enums.String_SList.GSlist
    is
       function Internal (Chooser : Gtk_File_Chooser) return System.Address;
-      pragma Import (C, Internal, "gtk_file_chooser_list_shortcut_folders");
+      pragma Import
+        (C, Internal, "gtk_file_chooser_list_shortcut_folders_utf8");
       L : Gtk.Enums.String_SList.GSlist;
    begin
       Gtk.Enums.String_SList.Set_Object (L, Internal (Chooser));
@@ -454,7 +456,8 @@
          Folder  : String;
          Error   : access Glib.Error.GError)
          return Gboolean;
-      pragma Import (C, Internal, "gtk_file_chooser_remove_shortcut_folder");
+      pragma Import
+        (C, Internal, "gtk_file_chooser_remove_shortcut_folder_utf8");
 
       Err : aliased Glib.Error.GError;
    begin
@@ -500,7 +503,7 @@
    is
       function Internal
         (Chooser  : Gtk_File_Chooser; Filename : String) return Gboolean;
-      pragma Import (C, Internal, "gtk_file_chooser_select_filename");
+      pragma Import (C, Internal, "gtk_file_chooser_select_filename_utf8");
    begin
       return Boolean'Val (Internal (Chooser, Filename & ASCII.NUL));
    end Select_Filename;
@@ -528,7 +531,7 @@
    is
       function Internal
         (Chooser  : Gtk_File_Chooser; Filename : String) return Gboolean;
-      pragma Import (C, Internal, "gtk_file_chooser_set_current_folder");
+      pragma Import (C, Internal, "gtk_file_chooser_set_current_folder_utf8");
    begin
       return Boolean'Val (Internal (Chooser, Filename & ASCII.NUL));
    end Set_Current_Folder;
@@ -600,7 +603,7 @@
    is
       function Internal
         (Chooser  : Gtk_File_Chooser; Filename : String) return Gboolean;
-      pragma Import (C, Internal, "gtk_file_chooser_set_filename");
+      pragma Import (C, Internal, "gtk_file_chooser_set_filename_utf8");
    begin
       return Boolean'Val (Internal (Chooser, Filename & ASCII.NUL));
    end Set_Filename;
@@ -726,7 +729,7 @@
      (Chooser  : Gtk_File_Chooser; Filename : String)
    is
       procedure Internal (Chooser  : Gtk_File_Chooser; Filename : String);
-      pragma Import (C, Internal, "gtk_file_chooser_unselect_filename");
+      pragma Import (C, Internal, "gtk_file_chooser_unselect_filename_utf8");
    begin
       Internal (Chooser, Filename & ASCII.NUL);
    end Unselect_Filename;
Index: glib-convert.adb
===================================================================
--- glib-convert.adb	(revision 174374)
+++ glib-convert.adb	(working copy)
@@ -200,7 +200,7 @@
         (URI      : String;
          Hostname : access chars_ptr;
          Error    : GError_Access) return chars_ptr;
-      pragma Import (C, Internal, "g_filename_from_uri");
+      pragma Import (C, Internal, "g_filename_from_uri_utf8");
 
       S   : constant chars_ptr := Internal (URI & ASCII.NUL, Hostname, Error);
       Str : constant String := Value (S);
@@ -224,7 +224,7 @@
          Bytes_Read    : System.Address := System.Null_Address;
          Bytes_Written : System.Address := System.Null_Address;
          Error         : GError_Access) return chars_ptr;
-      pragma Import (C, Internal, "g_filename_from_utf8");
+      pragma Import (C, Internal, "g_filename_from_utf8_utf8");
 
       S   : constant chars_ptr := Internal
         (UTF8_String, UTF8_String'Length, Error => Error);
@@ -248,7 +248,7 @@
         (URI      : String;
          Hostname : System.Address;
          Error    : GError_Access) return chars_ptr;
-      pragma Import (C, Internal, "g_filename_to_uri");
+      pragma Import (C, Internal, "g_filename_to_uri_utf8");
 
       S    : chars_ptr;
       Host : aliased constant String := Hostname & ASCII.NUL;
@@ -282,7 +282,7 @@
          Bytes_Read    : System.Address := System.Null_Address;
          Bytes_Written : System.Address := System.Null_Address;
          Error         : GError_Access) return chars_ptr;
-      pragma Import (C, Internal, "g_filename_to_utf8");
+      pragma Import (C, Internal, "g_filename_to_utf8_utf8");
 
       S   : constant chars_ptr := Internal
         (OS_String, OS_String'Length, Error => Error);
Index: glib-values.ads
===================================================================
--- glib-values.ads	(revision 174374)
+++ glib-values.ads	(working copy)
@@ -170,7 +170,8 @@
    --  also provided inside Gtk.Text_Iter.
 
 private
-   type GValue_Data is array (1 .. 2) of Guint64;
+
+   type GValue_Data is array (1 .. 20) of Guint8;
    type GValue is record
       g_type : GType := GType_Invalid;
       data   : GValue_Data;
Index: gtk-icon_theme.adb
===================================================================
--- gtk-icon_theme.adb	(revision 174374)
+++ gtk-icon_theme.adb	(working copy)
@@ -133,7 +133,7 @@
 
    function Get_Filename (Icon_Info : Gtk_Icon_Info) return String is
       function Internal (Icon_Info : Gtk_Icon_Info) return Chars_Ptr;
-      pragma Import (C, Internal, "gtk_icon_info_get_filename");
+      pragma Import (C, Internal, "gtk_icon_info_get_filename_utf8");
    begin
       --  returned value owned by gtk+
       return ICS.Value (Internal (Icon_Info));
@@ -184,7 +184,7 @@
       procedure Internal
         (Icon_Theme : System.Address;
          Path       : String);
-      pragma Import (C, Internal, "gtk_icon_theme_append_search_path");
+      pragma Import (C, Internal, "gtk_icon_theme_append_search_path_utf8");
    begin
       Internal (Get_Object (Icon_Theme), Path & ASCII.NUL);
    end Append_Search_Path;
@@ -281,7 +281,7 @@
         (Icon_Theme : System.Address;
          Path       : out chars_ptr_array_access;
          N_Elements : out Gint);
-      pragma Import (C, Internal, "gtk_icon_theme_get_search_path");
+      pragma Import (C, Internal, "gtk_icon_theme_get_search_path_utf8");
 
       P : chars_ptr_array_access;
       N : Gint;
@@ -474,7 +474,7 @@
       procedure Internal
         (Icon_Theme : System.Address;
          Path       : String);
-      pragma Import (C, Internal, "gtk_icon_theme_prepend_search_path");
+      pragma Import (C, Internal, "gtk_icon_theme_prepend_search_path_utf8");
    begin
       Internal (Get_Object (Icon_Theme), Path & ASCII.NUL);
    end Prepend_Search_Path;
@@ -523,7 +523,7 @@
         (Icon_Theme : System.Address;
          Path       : Chars_Ptr_Array;
          N_Elements : Gint);
-      pragma Import (C, Internal, "gtk_icon_theme_set_search_path");
+      pragma Import (C, Internal, "gtk_icon_theme_set_search_path_utf8");
 
       C : Chars_Ptr_Array := From_String_List (Path);
    begin
Index: gtk-accel_map.adb
===================================================================
--- gtk-accel_map.adb	(revision 174374)
+++ gtk-accel_map.adb	(working copy)
@@ -58,7 +58,7 @@
 
    procedure Save (File_Name : String) is
       procedure Internal (File_Name : String);
-      pragma Import (C, Internal, "gtk_accel_map_save");
+      pragma Import (C, Internal, "gtk_accel_map_save_utf8");
    begin
       Internal (File_Name & ASCII.NUL);
    end Save;
@@ -69,7 +69,7 @@
 
    procedure Load (File_Name : String) is
       procedure Internal (File_Name : String);
-      pragma Import (C, Internal, "gtk_accel_map_load");
+      pragma Import (C, Internal, "gtk_accel_map_load_utf8");
    begin
       Internal (File_Name & ASCII.NUL);
    end Load;
Index: gtk-rc.adb
===================================================================
--- gtk-rc.adb	(revision 174374)
+++ gtk-rc.adb	(working copy)
@@ -83,7 +83,7 @@
 
    procedure Add_Default_File (Filename : String) is
       procedure Internal (Filename : String);
-      pragma Import (C, Internal, "gtk_rc_add_default_file");
+      pragma Import (C, Internal, "gtk_rc_add_default_file_utf8");
 
    begin
       Internal (Filename & ASCII.NUL);
@@ -95,7 +95,7 @@
 
    procedure Set_Default_Files (Filenames : Chars_Ptr_Array) is
       procedure Internal (Filenames : Chars_Ptr_Array);
-      pragma Import (C, Internal, "gtk_rc_set_default_files");
+      pragma Import (C, Internal, "gtk_rc_set_default_files_utf8");
 
    begin
       Internal (Filenames + Null_Ptr);
@@ -137,7 +137,7 @@
 
    procedure Parse (Filename : String) is
       procedure Internal (Filename : String);
-      pragma Import (C, Internal, "gtk_rc_parse");
+      pragma Import (C, Internal, "gtk_rc_parse_utf8");
 
    begin
       Internal (Filename & ASCII.NUL);
Index: gdk-pixbuf.adb
===================================================================
--- gdk-pixbuf.adb	(revision 174374)
+++ gdk-pixbuf.adb	(working copy)
@@ -423,7 +423,7 @@
       function Internal
         (Filename : String;
          Error    : access GError) return System.Address;
-      pragma Import (C, Internal, "gdk_pixbuf_new_from_file");
+      pragma Import (C, Internal, "gdk_pixbuf_new_from_file_utf8");
 
       Err : aliased GError;
 
@@ -444,7 +444,7 @@
       function Internal
         (Filename : String;
          Error    : access GError) return Gdk_Pixbuf_Animation;
-      pragma Import (C, Internal, "gdk_pixbuf_animation_new_from_file");
+      pragma Import (C, Internal, "gdk_pixbuf_animation_new_from_file_utf8");
 
       Err : aliased GError;
 
Index: gtk-file_selection.adb
===================================================================
--- gtk-file_selection.adb	(revision 174374)
+++ gtk-file_selection.adb	(working copy)
@@ -154,7 +154,7 @@
       function Internal
         (File_Selection : System.Address)
          return Interfaces.C.Strings.chars_ptr;
-      pragma Import (C, Internal, "gtk_file_selection_get_filename");
+      pragma Import (C, Internal, "gtk_file_selection_get_filename_utf8");
 
    begin
       return Interfaces.C.Strings.Value
@@ -297,7 +297,7 @@
    is
       procedure Internal
         (File_Selection : System.Address; Filename : UTF8_String);
-      pragma Import (C, Internal, "gtk_file_selection_set_filename");
+      pragma Import (C, Internal, "gtk_file_selection_set_filename_utf8");
 
    begin
       Internal (Get_Object (File_Selection), Filename & ASCII.NUL);
@@ -356,7 +356,7 @@
       return GNAT.Strings.String_List
    is
       function Internal (Filesel : System.Address) return System.Address;
-      pragma Import (C, Internal, "gtk_file_selection_get_selections");
+      pragma Import (C, Internal, "gtk_file_selection_get_selections_utf8");
 
       function Get_Length (S : System.Address) return Integer;
       pragma Import (C, Get_Length, "ada_string_array_length");
Index: glib-module.adb
===================================================================
--- glib-module.adb	(revision 174374)
+++ glib-module.adb	(working copy)
@@ -54,7 +54,7 @@
       function Internal
         (File_Name : String;
          Flags     : Module_Flags) return G_Module;
-      pragma Import (C, Internal, "g_module_open");
+      pragma Import (C, Internal, "g_module_open_utf8");
 
    begin
       return Internal (File_Name & ASCII.NUL, Flags);
@@ -113,7 +113,7 @@
 
    function Module_Name (Module : G_Module) return String is
       function Internal (Module : G_Module) return chars_ptr;
-      pragma Import (C, Internal, "g_module_name");
+      pragma Import (C, Internal, "g_module_name_utf8");
 
    begin
       return Value (Internal (Module));
Index: gtk-ui_manager.adb
===================================================================
--- gtk-ui_manager.adb	(revision 174374)
+++ gtk-ui_manager.adb	(working copy)
@@ -85,7 +85,7 @@
         (Self     : System.Address;
          Filename : String;
          Error    : GError_Access) return Guint;
-      pragma Import (C, Internal, "gtk_ui_manager_add_ui_from_file");
+      pragma Import (C, Internal, "gtk_ui_manager_add_ui_from_file_utf8");
    begin
       return Internal (Get_Object (Self), Filename & ASCII.NUL, Error);
    end Add_UI_From_File;
Index: gtk-icon_factory.adb
===================================================================
--- gtk-icon_factory.adb	(revision 174374)
+++ gtk-icon_factory.adb	(working copy)
@@ -204,7 +204,7 @@
       procedure Internal
         (Source   : Gtk_Icon_Source;
          Filename : String);
-      pragma Import (C, Internal, "gtk_icon_source_set_filename");
+      pragma Import (C, Internal, "gtk_icon_source_set_filename_utf8");
 
    begin
       Internal (Source, Filename & ASCII.NUL);
@@ -543,7 +543,7 @@
    function Get_Filename (Source : Gtk_Icon_Source) return String is
       function Internal (Source : Gtk_Icon_Source)
          return Interfaces.C.Strings.chars_ptr;
-      pragma Import (C, Internal, "gtk_icon_source_get_filename");
+      pragma Import (C, Internal, "gtk_icon_source_get_filename_utf8");
       --  Return value must not be freed.
    begin
       return Value (Internal (Source));
Index: gtk-window.adb
===================================================================
--- gtk-window.adb	(revision 174374)
+++ gtk-window.adb	(working copy)
@@ -858,7 +858,7 @@
         (Window   : System.Address;
          Filename : String)
          return Gboolean;
-      pragma Import (C, Internal, "gtk_window_set_icon_from_file");
+      pragma Import (C, Internal, "gtk_window_set_icon_from_file_utf8");
    begin
       return Boolean'Val
         (Internal (Get_Object (Window), Filename & ASCII.NUL));
@@ -1016,7 +1016,8 @@
 
    function Set_Default_Icon_From_File (Filename : String) return Boolean is
       function Internal (Filename : String) return Gboolean;
-      pragma Import (C, Internal, "gtk_window_set_default_icon_from_file");
+      pragma Import
+        (C, Internal, "gtk_window_set_default_icon_from_file_utf8");
    begin
       return Boolean'Val (Internal (Filename & ASCII.NUL));
    end Set_Default_Icon_From_File;
Index: generated/gtk-image.adb
===================================================================
--- generated/gtk-image.adb	(revision 174374)
+++ generated/gtk-image.adb	(working copy)
@@ -204,7 +204,7 @@
    is
       function Internal
          (Filename : Interfaces.C.Strings.chars_ptr) return System.Address;
-      pragma Import (C, Internal, "gtk_image_new_from_file");
+      pragma Import (C, Internal, "gtk_image_new_from_file_utf8");
       Tmp_Filename : Interfaces.C.Strings.chars_ptr := New_String (Filename);
       Tmp_Return   : System.Address;
    begin
@@ -497,7 +497,7 @@
       procedure Internal
          (Self     : System.Address;
           Filename : Interfaces.C.Strings.chars_ptr);
-      pragma Import (C, Internal, "gtk_image_set_from_file");
+      pragma Import (C, Internal, "gtk_image_set_from_file_utf8");
       Tmp_Filename : Interfaces.C.Strings.chars_ptr := New_String (Filename);
    begin
       Internal (Get_Object (Self), Tmp_Filename);
